pkg/scheduler/plugins/minruntime/resolver.go

```diff
@@ -0,0 +1,241 @@
+			if found {
+				currentQueue = parentQueue
+				continue
+			}
+		}
+
+		// Break if no more parent queues
+		break
+	}
+
+	// If no preempt-min-runtime is set in the queue tree, use default
+	mr.preemptMinRuntimeCache[queue.UID] = *mr.defaultPreemptMinRuntime
+	return *mr.defaultPreemptMinRuntime
+}
+
+// getReclaimMinRuntime resolves min-runtime for reclaims
+// Depending on the resolveMethod, it will use either:
+// 1. queue: Starting from the leaf-queue, walk the tree (similar to preempt)
+// 2. lca: Use the lowest common ancestor approach
+func (mr *minruntimePlugin) getReclaimMinRuntime(
+	resolveMethod string,
+	preemptorQueue *queue_info.QueueInfo,
+	preempteeQueue *queue_info.QueueInfo,
+) metav1.Duration {
+	if preemptorQueue == nil || preempteeQueue == nil {
+		return *mr.defaultReclaimMinRuntime
+	}
+
+	if minRuntime, ok := mr.reclaimMinRuntimeCache[preemptorQueue.UID][preempteeQueue.UID]; ok {
+		return minRuntime
+	}
+
+	// If method is LCA, use the LCA-based algorithm
+	if resolveMethod == resolveMethodLCA {
+		return mr.getReclaimMinRuntimeLCA(preemptorQueue, preempteeQueue)
+	}
+
+	// Otherwise, use queue-based approach (similar to preempt)
+	// Start with the provided leaf queue
+	currentQueue := preempteeQueue
+
+	// Walk up the tree until we find a set reclaim-min-runtime or reach the root
+	for currentQueue != nil {
+		if currentQueue.ReclaimMinRuntime != nil {
+			return *currentQueue.ReclaimMinRuntime
+		}
+
+		// Move to parent queue if it exists
+		if currentQueue.ParentQueue != "" {
+			parentQueue, found := mr.queues[currentQueue.ParentQueue]
+			if found {
+				currentQueue = parentQueue
+				continue
+			}
+		}
+
+		// Break if no more parent queues
+		break
+	}
+
+	// If no reclaim-min-runtime is set in the queue tree, use default
+	return *mr.defaultReclaimMinRuntime
+}
+
+// getReclaimMinRuntimeLCA resolves min-runtime for reclaims using the LCA method
+// 1. Resolve the lowest common ancestor (LCA) between the leaf-queues of preemptor and preemptee
+// 2. Walk 1 step down to the child of the LCA that is an ancestor to the preemptee's leaf queue (or is the leaf queue)
+// 3. Use the reclaim-min-runtime from this queue, if it is set. Otherwise move back up towards root of tree
+//
+// Note: Top-level queues (with empty parent) are treated as siblings with an implicit shadow parent
+func (mr *minruntimePlugin) getReclaimMinRuntimeLCA(
+	preemptorQueue *queue_info.QueueInfo,
+	preempteeQueue *queue_info.QueueInfo,
+) metav1.Duration {
+	if preemptorQueue == nil || preempteeQueue == nil {
+		return *mr.defaultReclaimMinRuntime
+	}
+
+	// Get hierarchy paths for both queues
+	preemptorPath := mr.getQueueHierarchyPath(preemptorQueue)
+	preempteePath := mr.getQueueHierarchyPath(preempteeQueue)
+
+	// Handle case where both queues are from different top-level hierarchies
+	// Consider them as siblings with a shadow parent
+	if len(preemptorPath) > 0 && len(preempteePath) > 0 {
```

Comment:
Here, instead of nesting importing logic, just handle the case of missing preemptor path early and get it out of the way:
```suggestion
	if len(preemptorPath) == 0 {
		// Handle error - return default value? return error?
	}
	// Do the same for preempteePath (we can combine this to a single `if` statement, but then we will loose some info for the error
	
	// proceed to happy flow
```